pkg/flags/flags.go

```diff
@@ -24,40 +24,70 @@ const (
 	namespace  = "namespace"
 )
 
-// AddTektonOptions amends command to add flags required to initialise a cli.Param
-func AddTektonOptions(cmd *cobra.Command) {
+type Options struct {
+	Required bool
+}
+
+// Initializer accepts a cli.Param and sets/initializes it based on flags in cobra.Command
+type Initializer func(p cli.Params, cmd *cobra.Command) error
+
+// RunE is the type for cobra.RunE
+type RunE func(cmd *cobra.Command, args []string) error
+
+// InitParams initialises cli.Params
+func InitParams(p cli.Params, ilist ...Initializer) RunE {
+	return func(cmd *cobra.Command, args []string) error {
+		for _, fn := range ilist {
+			if err := fn(p, cmd); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+}
+
+// FromKubeConfig adds an optional --kubeconfig to cmd
+func FromKubeConfig(cmd *cobra.Command, o Options) Initializer {
 	cmd.PersistentFlags().StringP(
 		kubeConfig, "k", "",
 		"kubectl config file (default: $HOME/.kube/config)")
-
-	cmd.PersistentFlags().StringP(
-		namespace, "n", "",
-		"namespace to use (mandatory)")
-
-	cmd.MarkPersistentFlagRequired(namespace)
+	setOptions(cmd, o)
```

Comment:
how about doing it `o.for(cmd)` or `o.set(cmd)` ? :thinking:

Reply:
what advantage do you see with that approach?